dashboard/src/actions/authActions.js

```diff
@@ -2,203 +2,70 @@ import * as APP_ROUTES from "utils/routeConstants";
-      });
-      dispatch({ type: TYPES.COMPLETED });
+/**
+ * Wait for the Pbench Server endpoints to be loaded.
+ * @param {getState} getState object.
+ * @return {promise} promise object
+ */
+export async function waitForEndpoints(getState) {
+  const waitStart = Date.now();
+  /**
+   * Settle the wait-for-endpoints promise.
+   * @param {resolve} resolve object.
+   * @param {reject} reject object
+   */
+  function check(resolve, reject) {
+    if (Object.keys(getState().apiEndpoint.endpoints).length !== 0) {
+      resolve("Endpoints loaded");
+    } else if (Date.now() - waitStart > CONSTANTS.MAX_WAIT_MS) {
+      reject(new Error("Timed out waiting for endpoints request"));
+    } else {
+      setTimeout(check, 250, resolve, reject);
     }
-};
-
-export const makeLoginRequest =
-  (details, navigate) => async (dispatch, getState) => {
-    try {
-      dispatch({ type: TYPES.LOADING });
-      // empty the alerts
-      dispatch({
-        type: TYPES.USER_NOTION_ALERTS,
-        payload: [],
-      });
-      const endpoints = getState().apiEndpoint.endpoints;
-      const response = await API.post(endpoints?.api?.login, {
-        ...details,
-      });
-      if (response.status === 200 && Object.keys(response.data).length > 0) {
-        const keepUser = getState().userAuth.keepLoggedIn;
-        const expiryTime = keepUser
-          ? CONSTANTS.EXPIRY_KEEPUSER_DAYS
-          : CONSTANTS.EXPIRY_DEFAULT_DAYS;
-        Cookies.set("isLoggedIn", true, { expires: expiryTime });
-        Cookies.set("token", response.data?.auth_token, {
-          expires: expiryTime,
-        });
-        Cookies.set("username", response.data?.username, {
-          expires: expiryTime,
-        });
-        const loginDetails = {
-          isLoggedIn: true,
-          token: response.data?.auth_token,
-          username: response.data?.username,
-        };
-        await dispatch({
-          type: TYPES.SET_LOGIN_DETAILS,
-          payload: loginDetails,
-        });
-
-        navigate(APP_ROUTES.OVERVIEW);
+  }
+  return new Promise((resolve, reject) => check(resolve, reject));
+}
 
-        dispatch(showToast(SUCCESS, "Logged in successfully!"));
-      }
-      dispatch({ type: TYPES.COMPLETED });
-    } catch (error) {
-      const alerts = getState().userAuth.alerts;
-      let alert = {};
-      if (error?.response) {
-        alert = {
-          title: error?.response?.data?.message,
-          key: uid(),
-        };
-        dispatch(toggleLoginBtn(true));
-      } else {
-        alert = {
-          title: error?.message,
-          key: uid(),
-        };
-        dispatch({ type: TYPES.NETWORK_ERROR });
-      }
-      alerts.push(alert);
-      dispatch({
-        type: TYPES.USER_NOTION_ALERTS,
-        payload: alerts,
-      });
-      dispatch({ type: TYPES.COMPLETED });
-    }
-  };
+// Perform some house keeping when the user logs in
+export const authCookies = () => async (dispatch, getState) => {
```

Comment:
This is another example of "a function returning a thunk" which could (or _should_) just **_be_** the thunk.  That is, if you look at the caller of this function, it calls it with no arguments and passes the returned value to `dispatch()`; instead, the caller should just pass `authCookies` itself (without calling it) to `dispatch()`, and its definition should be:
```javascript
export const authCookies = async (dispatch, getState) => {
```
(i.e., the same with the extra level of "callableness" removed).